spec/custom-gutter-component-spec.coffee

```diff
@@ -0,0 +1,132 @@
+CustomGutterComponent = require '../src/custom-gutter-component'
+Gutter = require '../src/gutter'
+
+describe "CustomGutterComponent", ->
+  [customGutterComponent, gutter] = []
+
+  beforeEach ->
+    mockGutterContainer = {}
+    gutter = new Gutter(mockGutterContainer, {name: 'test-gutter'})
+    customGutterComponent = new CustomGutterComponent({gutter})
+
+  it "creates a gutter DOM node with only an empty 'custom-decorations' child node when it is initialized", ->
+    expect(customGutterComponent.getDomNode().classList.contains('gutter')).toBe true
+    expect(customGutterComponent.getDomNode().getAttribute('gutter-name')).toBe 'test-gutter'
+    expect(customGutterComponent.getDomNode().children.length).toBe 1
+    decorationsWrapperNode = customGutterComponent.getDomNode().children.item(0)
+    expect(decorationsWrapperNode.classList.contains('custom-decorations')).toBe true
+
+  it "makes its view accessible from the view registry", ->
+    expect(customGutterComponent.getDomNode()).toBe atom.views.getView(gutter)
+
+  it "hides its DOM node when ::hideNode is called, and shows its DOM node when ::showNode is called", ->
```

Comment:
Who calls `::hideNode`? I thought it was called internally by the component when the state changes. I think I'd prefer to just go straight to an integration test, since it's hard to imagine this code ever working without the presenter and the model.